### ISSUE TITLE
Hierarchical layout doesn't place self loop ports properly

### ISSUE THREAD

**Issue by le-cds** [eclipse/elk#552]
Consider this graph:

```
hierarchyHandling: INCLUDE_CHILDREN
node N1 {
	port P1
	port P2
	port P3
	node N2 {
		layout [ size: 20, 20 ]
	}
	edge E2: P1 -> N2
}
edge E1: N1.P3 -> N1.P2
```

The following layout is produced:

![problem](https://user-images.githubusercontent.com/7465584/78904278-91ddff80-7a7c-11ea-90e3-9f5e1e56b46a.png)

The self loop ports are not placed properly.

**Comment by le-cds** [comment]
Note to self: The innermost `LGraph` has the expected number of three external port dummies as we begin executing `ElkLayered.hierarchicalLayout(LGraph, IELKProgressMonitor)`.

**Comment by le-cds** [comment]
The problem was a fix for #352. Since this involved unintended side effects across hierarchy boundaries, this might be worth describing for future reference. What (roughly) happened was this.

Hierarchy mode was enabled. Since crossing minimization (P3) operates on the whole hierarchy, the algorithm's pipeline looked like this:

1. Pre-P3 processors for inner graph.
1. Pre-P3 processors for outer graph.
1. P3 for whole hierarchy.
1. Post-P3 processors for inner graph.
1. Post-P3 processors for outer graph.

Since the inner graph had a connection the parent, it had an external port. Thus, all of its three ports were represented by external port dummies, as expected. As part of step 2, ELK Layered ran the `SelfLoopPreProcessor` on the outer graph, which in fact found that the graph's only node had, in fact, a self loop. Its port order was not fixed, so the processor went ahead and removed the self loop's ports. The fix for #352 now caused the corresponding external port dummies in the inner graph to be removed as well.

In step 4, layout on the inner graph (which only had a single external port dummy left now) continued. The single dummy was positioned, and at the very end of step 4, its position was applied to the corresponding port and the node's port constraints set to `FIXED_POS`. The y positions of the two self loop ports were never touched, so they remained at 0, and due to the new port constraints that's where they remained.

The fix to this issue is to not hide any self loop ports if the node contains another graph. At the end of the inner graph's layout, the ports have a fixed position which the outer layout run can then work with just fine.

I do believe that this will stop working once we introduce further processors or phases that work on the root node. This fix relies on the fact that the inner graph's layout runs to completion before the outer graph's layout picks up after P3. Another root processor might cause the assumption to fail.
